src/dbnode/x/xio/segment_reader.go

```diff
@@ -48,28 +50,29 @@ func (sr *segmentReader) Read(b []byte) (int, error) {
 	if len(b) == 0 {
 		return 0, nil
 	}
-	var head, tail []byte
-	if b := sr.segment.Head; b != nil {
-		head = b.Bytes()
+
+	if b := sr.segment.Head; b != nil && sr.lazyHead == nil {
```

Comment:
This will have to be `len(sr.lazyHead) > 0` now that the reset doesn't nil these

Reply:
So I'm not sure I follow this condition - `lazyHead` starts as `nil` and so this condition will never be met until it gets the initial set to `segment.Head.Bytes()`, but that happens only if this condition is true. Aren't we trying to say:
```
if not yet set {
    set
}
```